packages/reporter/src/reporters/gitlab/GitLabCodeQualityReporter.spec.ts

```diff
@@ -0,0 +1,94 @@
+import { GitLabCIArtifacts, GITLAB_CI_ARTIFACTS, GITLAB_CONFIG } from './api';
+import { fullyDescribedIssue } from '../../__fixtures__/issues';
+import { TEST_FILE_PATH_RESOLVER, TestFilePathResolver } from '../../utils';
+import { CodeQualityReport } from './types/CodeQualityReport';
+import { Issue, Scan } from '@sectester/scan';
+import { container } from 'tsyringe';
+import {
+  anything,
+  capture,
+  instance,
+  mock,
+  reset,
+  verify,
+  when
+} from 'ts-mockito';
+
+describe('GitLabCodeQualityReporter', () => {
+  let reporter: GitLabCodeQualityReporter;
+  const mockedScan = mock<Scan>();
+  const mockedGitLabCIArtifacts = mock<GitLabCIArtifacts>();
+  const mockedTestFilePathResolver = mock<TestFilePathResolver>();
+
+  const mockConfig = {};
+
+  beforeEach(() => {
+    container.clearInstances();
+
+    container.register(GITLAB_CONFIG, { useValue: mockConfig });
+    container.register(GITLAB_CI_ARTIFACTS, {
+      useValue: instance(mockedGitLabCIArtifacts)
+    });
+    container.register(TEST_FILE_PATH_RESOLVER, {
+      useValue: instance(mockedTestFilePathResolver)
+    });
+
+    when(mockedTestFilePathResolver.getTestFilePath()).thenReturn(
+      'test.spec.ts'
+    );
+
+    reporter = container.resolve(GitLabCodeQualityReporter);
+  });
+
+  afterEach(() => {
+    reset<Scan | GitLabCIArtifacts | TestFilePathResolver>(
+      mockedScan,
+      mockedGitLabCIArtifacts,
+      mockedTestFilePathResolver
+    );
+  });
+
+  describe('report', () => {
+    it('should not send code quality report when there are no issues', async () => {
+      when(mockedScan.issues()).thenResolve([]);
+
+      await reporter.report(instance(mockedScan));
+
+      verify(
+        mockedGitLabCIArtifacts.writeCodeQualityReport(anything())
+      ).never();
+    });
+
+    it('should send code quality report when there are issues', async () => {
+      when(mockedScan.issues()).thenResolve([fullyDescribedIssue] as Issue[]);
+      when(
+        mockedGitLabCIArtifacts.writeCodeQualityReport(anything())
+      ).thenResolve();
+
+      await reporter.report(instance(mockedScan));
+
+      verify(mockedGitLabCIArtifacts.writeCodeQualityReport(anything())).once();
+
+      const [actualReport]: [CodeQualityReport, ...any[]] = capture(
+        mockedGitLabCIArtifacts.writeCodeQualityReport
+      ).last();
+
+      expect(actualReport).toHaveLength(1);
+      expect(actualReport[0]).toEqual({
+        description: `${fullyDescribedIssue.name} vulnerability found at ${fullyDescribedIssue.originalRequest.method.toUpperCase()} ${fullyDescribedIssue.originalRequest.url}`,
+        fingerprint: expect.any(String),
+        check_name: fullyDescribedIssue.name,
+        severity: 'major',
+        raw_details: JSON.stringify(fullyDescribedIssue, null, 2),
+        location: {
+          path: 'test.spec.ts',
+          lines: {
+            begin: 1
+          }
+        }
+      });
+    });
+  });
```

Comment:
Assert the entire response

```suggestion
      expect(actualReport).toEqual([{
        description: `${fullyDescribedIssue.name} vulnerability found at ${fullyDescribedIssue.originalRequest.method.toUpperCase()} ${fullyDescribedIssue.originalRequest.url}`,
        fingerprint: expect.any(String),
        check_name: fullyDescribedIssue.name,
        severity: 'major',
        raw_details: JSON.stringify(fullyDescribedIssue, null, 2),
        location: {
          path: 'test.spec.ts',
          lines: {
            begin: 1
          }
        }
      });
    });
  }]);
```